pkg/resources/database_state_upgraders.go

```diff
@@ -0,0 +1,39 @@
+package resources
+
+import (
+	"context"
+)
+
+func v092DatabaseStateUpgrader(ctx context.Context, rawState map[string]any, meta any) (map[string]any, error) {
+	if rawState == nil {
+		return rawState, nil
+	}
+
```

Comment:
Hmm, good question, because this could also work if they would upgrade into the standard database with from_share commented. Then to make it right, they would do the migration to the correct type with state manipulation (then they would change the type to `snowflake_shared_database` and would change the internal config to correct `from_share`), but on 0.92.0 instead of between the versions. But maybe throwing an error would be better because it would prevent users from using incorrect database type.

Reply:
exactly, so we agree to add this? (it may be the follow-up PR